torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -293,6 +287,31 @@ void DebugInfoWriter::write(const std::string& ncclTrace) {
   LOG(INFO) << "Finished writing NCCLPG debug info to " << filename_;
 }
 
+DebugInfoWriter& DebugInfoWriter::getWriter(int rank) {
+  if (writer_ == nullptr) {
```

Comment:
we made `hasWriterRegistered_` atomic. Yet we are still checking `writer_` here.

Reply:
Well I can change this check to check the atomic variable but that does not fully protect the setWriter.